main.tf

```diff
@@ -354,11 +355,13 @@ resource "ibm_container_vpc_worker_pool" "pool" {
     create = "2h"
   }
 
+  # The default workerpool has to be imported as it will already exist on cluster create
+  import_on_create = each.value.pool_name == "default" ? true : null
```

Comment:
The issue with forcing `import_on_create` on the default worker pool is that if this resource has to be replaced (change of flavor for example), it will fail.

So maybe we could add a new field `import_on_create` to the `var.worker_pools`, then
```
import_on_create = each.value.pool_name == "default" ? each.value.import_on_create == null ? true : each.value.import_on_create : null
```

That way if the default workerpool needs replacement because of a flavor change, the user can set `import_on_create = false` and do the flavor change as well

Reply:
This is how it was previously done when the original testing comment was created
https://github.com/terraform-ibm-modules/terraform-ibm-base-ocp-vpc/pull/499/files/533e7fcc9434d704ae5f6f5254c7d70f7432d469#diff-dc46acf24afd63ef8c556b77c126ccc6e578bc87e3aa09a931f33d9bf2532fbbR357

Reply:
This is getting messy.... Feels like we're accumulating workarounds with the introduction of import_on_create in provider on the edge cases, as that field is not fundamentally aligned with the service.

Anyways, I think we just add a new field for that use case at the top of the module. Name it allow_default_worker_pool_replacement and if set to true then import_on_create is set to false for the default worker pool.

Reply:
Its really getting complex, the user should be aware that they need to set allow_default_worker_pool_replacement  to true when they wish to make any change to the default_worker_pool which needs the re-creation of the worker-pool.

But again, in case of a cluster with only default pool, we wont be able to make any changes to worker-pool which needs a re-creation, it will fail at trying to destroy since the number of nodes can't be 0